Now it passes further and stops with this error instead:
```
ninja: Entering directory `build'                                                                                                                                                                                                              
[8/1481] Compiling C object librz/util/msys-rz_util-0.2.0-git.dll.p/print_code.c.o                                                                                                                                                             
FAILED: librz/util/msys-rz_util-0.2.0-git.dll.p/print_code.c.o                                                                                                                                                                                 
cc -Ilibrz/util/msys-rz_util-0.2.0-git.dll.p -I. -I.. -I../librz/include -I../shlr/spp -Isubprojects/sdb -I../subprojects/sdb -Isubprojects/sdb/src -I../subprojects/sdb/src -Isubprojects/zlib-1.2.11 -I../subprojects/zlib-1.2.11 -fdiagnosti
cs-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -g --std=gnu99 -fvisibility=hidden -D_GNU_SOURCE -Werror=sizeof-pointer-memaccess -pthread -DRZ_PLUGIN_INCORE=1 -MD -MQ librz/util/msys-rz_util-0.2.0-git.dll.p/print_code.c.o
 -MF librz/util/msys-rz_util-0.2.0-git.dll.p/print_code.c.o.d -o librz/util/msys-rz_util-0.2.0-git.dll.p/print_code.c.o -c ../librz/util/print_code.c                                                                                          
In file included from ../librz/include/rz_io.h:8,                                                                                                                                                                                              
                 from ../librz/include/rz_util/rz_print.h:7,                                                                                                                                                                                   
                 from ../librz/util/print_code.c:4:                                                                                                                                                                                            
../librz/include/rz_socket.h:70:21: error: field ‘sa’ has incomplete type                                                                                                                                                                      
   70 |  struct sockaddr_in sa;                                                                                                                                                                                                                
      |                     ^~                                                                                                                                                                                                                 
[11/1481] Compiling C object librzspp.a.p/shlr_spp_spp.c.o                                                                                                                                                                                     
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
  366 | }                                                                                                                                                                                                                                      
      | ^                                                                                                                                                                                                                                      
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
../shlr/spp/spp.c:366:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]                                                                                                                             
[16/1481] Compiling C object librz/util/msys-rz_util-0.2.0-git.dll.p/buf.c.o                                                                                                                                                                   
FAILED: librz/util/msys-rz_util-0.2.0-git.dll.p/buf.c.o                                                                                                                                                                                        
cc -Ilibrz/util/msys-rz_util-0.2.0-git.dll.p -I. -I.. -I../librz/include -I../shlr/spp -Isubprojects/sdb -I../subprojects/sdb -Isubprojects/sdb/src -I../subprojects/sdb/src -Isubprojects/zlib-1.2.11 -I../subprojects/zlib-1.2.11 -fdiagnosti
cs-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -g --std=gnu99 -fvisibility=hidden -D_GNU_SOURCE -Werror=sizeof-pointer-memaccess -pthread -DRZ_PLUGIN_INCORE=1 -MD -MQ librz/util/msys-rz_util-0.2.0-git.dll.p/buf.c.o -MF li
brz/util/msys-rz_util-0.2.0-git.dll.p/buf.c.o.d -o librz/util/msys-rz_util-0.2.0-git.dll.p/buf.c.o -c ../librz/util/buf.c                                                                                                                      
In file included from ../librz/include/rz_io.h:8,                                                                                                                                                                                              
                 from ../librz/util/buf.c:5:                                                                                                                                                                                                   
../librz/include/rz_socket.h:70:21: error: field ‘sa’ has incomplete type                                                                                                                                                                      
   70 |  struct sockaddr_in sa;                                                                                                                                                                                                                
      |                     ^~                                                                                                                                                                                                                 
[17/1481] Compiling C object librz/util/msys-rz_util-0.2.0-git.dll.p/cache.c.o                                                                                                                                                                 
ninja: build stopped: subcommand failed.                                                                                                                                                                                                       
```

--- a/librz/include/rz_types.h
+++ b/librz/include/rz_types.h
@@ -603,7 +603,7 @@ typedef enum {
 	RZ_SYS_ARCH_RISCV
 } RzSysArch;
 
-#if HAVE_CLOCK_NANOSLEEP && CLOCK_MONOTONIC && (__linux__ || (__FreeBSD__ && __FreeBSD_version >= 1101000) || (__NetBSD__ && __NetBSD_Version__ >= 700000000))
+#if HAVE_CLOCK_NANOSLEEP && defined(CLOCK_MONOTONIC) && (__linux__ || (__FreeBSD__ && __FreeBSD_version >= 1101000) || (__NetBSD__ && __NetBSD_Version__ >= 700000000))
 #define HAS_CLOCK_NANOSLEEP 1
 #else
 #define HAS_CLOCK_NANOSLEEP 0

--- a/librz/include/rz_util/rz_sys.h
+++ b/librz/include/rz_util/rz_sys.h
@@ -132,11 +132,6 @@ RZ_API int rz_sys_open(const char *path, int perm, int mode);
 RZ_API FILE *rz_sys_fopen(const char *path, const char *mode);
 RZ_API int rz_sys_truncate_fd(int fd, ut64 length);
 RZ_API int rz_sys_truncate(const char *file, int sz);
-#if __WINDOWS__
-RZ_API HANDLE rz_sys_opendir(const char *path, WIN32_FIND_DATAW *entry);
-#else
-RZ_API DIR *rz_sys_opendir(const char *path);
-#endif
 RZ_API int rz_sys_cmdbg(const char *cmd);
 RZ_API int rz_sys_cmdf(const char *fmt, ...) RZ_PRINTF_CHECK(1, 2);
 RZ_API char *rz_sys_cmd_str(const char *cmd, const char *input, int *len);

--- a/librz/util/sys.c
+++ b/librz/util/sys.c
@@ -198,12 +198,31 @@ RZ_API void rz_sys_exit(int status, bool nocleanup) {
 	}
 }
 
+#if __WINDOWS__
+static HANDLE sys_opendir(const char *path, WIN32_FIND_DATAW *entry) {
+	rz_return_val_if_fail(path, NULL);
+	wchar_t dir[MAX_PATH];
+	wchar_t *wcpath = 0;
+	if (!(wcpath = rz_utf8_to_utf16(path))) {
+		return NULL;
+	}
+	swprintf(dir, MAX_PATH, L"%ls\\*.*", wcpath);
+	free(wcpath);
+	return FindFirstFileW(dir, entry);
+}
+#else
+static DIR *sys_opendir(const char *path) {
+	rz_return_val_if_fail(path, NULL);
+	return opendir(path);
+}
+#endif
+
 RZ_API RzList *rz_sys_dir(const char *path) {
 	RzList *list = NULL;
 #if __WINDOWS__
 	WIN32_FIND_DATAW entry;
 	char *cfname;
-	HANDLE fh = rz_sys_opendir(path, &entry);
+	HANDLE fh = sys_opendir(path, &entry);
 	if (fh == INVALID_HANDLE_VALUE) {
 		//IFDGB eprintf ("Cannot open directory %ls\n", wcpath);
 		return list;
@@ -220,7 +239,7 @@ RZ_API RzList *rz_sys_dir(const char *path) {
 	FindClose(fh);
 #else
 	struct dirent *entry;
-	DIR *dir = rz_sys_opendir(path);
+	DIR *dir = sys_opendir(path);
 	if (dir) {
 		list = rz_list_new();
 		if (list) {
@@ -1944,24 +1963,7 @@ RZ_API int rz_sys_kill(int pid, int sig) {
 #endif
 	return -1;
 }
-#if __WINDOWS__
-RZ_API HANDLE rz_sys_opendir(const char *path, WIN32_FIND_DATAW *entry) {
-	rz_return_val_if_fail(path, NULL);
-	wchar_t dir[MAX_PATH];
-	wchar_t *wcpath = 0;
-	if (!(wcpath = rz_utf8_to_utf16(path))) {
-		return NULL;
-	}
-	swprintf(dir, MAX_PATH, L"%ls\\*.*", wcpath);
-	free(wcpath);
-	return FindFirstFileW(dir, entry);
-}
-#else
-RZ_API DIR *rz_sys_opendir(const char *path) {
-	rz_return_val_if_fail(path, NULL);
-	return opendir(path);
-}
-#endif
+
 RZ_API bool rz_sys_stop(void) {
 #if __UNIX__
 	return !rz_sys_kill(0, SIGTSTP);

